packages/webdriverio/tests/commands/browser/url.test.js

```diff
@@ -27,14 +27,39 @@ describe('url', () => {
             .toEqual({ url: 'http://foobar.com/foobar' })
     })
 
-    it('should throw an exception when a non-string value passed in', async () => {
+    it('should throw an exception when a non-string value passed in path', async () => {
         try {
             await browser.url(true)
         } catch (e) {
             expect(e.message).toContain('command needs to be type of string')
         }
     })
 
+    it('should throw an exception when a non-function value passed in inject', async () => {
+        try {
+            await browser.url('/', { inject: 123 })
+        } catch (e) {
+            expect(e.message).toContain('command needs to be type of function')
+        }
+    })
+
+    it('should continue the navigation if inject parameter is not passed', async () => {
```

Comment:
this test needs a check that puppeteer wasn't used

Reply:
@christian-bromann I believe I need help in this. Help in how to approach is fine.

Reply:
you need to mock out Puppeteer first:
```js
const pageMock = { evaluate: jest.fn() }
browser.getPuppeteer = jest.fn().mockReturnValue({
    pages: [pageMock]
})
```

then assert if the page mock was called twice:

```js
expect(pageMock.evaluate).toBeCalledTimes(2)
```